CIP-0068/README.md

```diff
@@ -0,0 +1,223 @@
+
+To find the metadata for the `user token` you need to look for the output, where the `reference NFT` is locked in. How this is done concretely will become clear below. Moreover, this output contains a datum, which holds the metadata. The advantage of this approach is that the issuer of the assets can decide how the transaction output with the `reference NFT` is locked and further handled. If the issuer wants complete immutable metadata, the `reference NFT` can be locked at the address of an unspendable script. Similarly, if the issuer wants the NFTs/FTs to evolve or wants a mechanism to update the metadata, the `reference NFT` can be locked at the address of a script with arbitrary logic that the issuer decides.
+
+Lastly and most importantly, with this construction, the metadata can be used by a PlutusV2 script with the use of reference inputs [CIP-0031](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0031). This will drive further innovation in the token space. 
+
+## Specification
+
+This is the registered `asset_name_label` value
+
+| asset_name_label            | class        | description                                               |
+| --------------------------- | ------------ | --------------------------------------------------------- |
+| 100                         | NFT          | Reference NFT locked at a script containing the datum     |
+
+For a correct relationship between the `user token` and the `reference NFT` a few conditions **must** be met.
+- The `user token` and `reference NFT` **must** be under the same policy ID. 
+- For a specific `user token` there **must** exist exactly **one** `reference NFT`
+- The `user token` and associated `reference NFT` **must** follow the standard naming pattern. The asset name of both assets is prefixed with its respective `asset_name_label` followed by a pattern defined by the asset class (e.g. asset_name_label 222)
+
+Some remarks about the above,
+1. The `user token` and `reference NFT` do not need to be minted in the same transaction. The order of minting is also not important.
+2. It may be the case that there can be multiple `user tokens` referencing the same `reference NFT`. More explicit with this standard, if a specific `user token` has a quantity bigger than 1, then they all reference the same `reference NFT`.
+
+The datum in the output with the `reference NFT` contains the metadata at the first field of the constructor 0. The version number is at the second field of this constructor:
+```
+big_int = int / big_uint / big_nint
+big_uint = #6.2(bounded_bytes)
+big_nint = #6.3(bounded_bytes)
+
+metadata =
+  { * metadata => metadata }
+  / [ * metadata ]
+  / big_int
+  / bounded_bytes
+  
+version = big_int
+
+datum = #6.121([metadata, version])
```

Comment:
Note: while reviewing [another CIP](https://github.com/cardano-foundation/CIPs/pull/249) that proposes an extension to CIP-0025, it occurred to me that CIP-0025 does not currently provide a nice extension mechanism. While there's now a version number, it forces every version increment to support all the newly introduce 'features'.

Rather, I would imagine a restricted set of core features and a flexible extension mechanisms which allows people to extend the standard to new use-cases while letting downstream consumers chose which extensions they intend to support. This could very much be CIP-driven, for example an extra field: `extensions` or `plugins` listing all the extensions used by metadata. A consumer service processing the metadata and not being aware of one of the required extension could therefore stop and error instead of showing potentially wrong metadata. 
Alternatively, each extensions could be marked as "required" or not (in the same spirit as it is done for vocabularies on JSON-schemas); as some extensions may only provide additional details (like, say, extra optional information) but ignoring them wouldn't affect the core interpretation of the metadata.

Reply:
@KtorZ, thank you for reviewing this CIP. 

> I'd hope CIP-0068 to resolve that short-coming.

I agree, and I think it does resolve these. Note that this CIP keeps things flexible with CIP 67 [(1)](https://github.com/cardano-foundation/CIPs/pull/298). The goal was to create a flexible framework for all kind of tokens, not just NFT's. I think the main why reason this is achieved, is because data is spread out and localized. 

@alessandrokonrad made the following diagram that shows how flexible this standard could be.
![Standard_1](https://user-images.githubusercontent.com/107037423/184830018-4d1f1a76-93fe-4bfb-b00d-cfca12de3eb4.svg)

Here there is one main minting policy script which can mint the main token. These could be FT's, NFT, fractionalized NFT, or any other token standard that might come up. Each standard is marked with `(xxx)` (though the delimiter is arbitrary at this point). Separate to this minting policy script, there is a registry which can introduce new standards under the main minting policy, in the example a royalty standard is made. Other standards/extensions can easily be introduced and made available under the main policy ID via the registry.

Reply:
> This could very much be CIP-driven, for example an extra field: `extensions` or `plugins` listing all the extensions used by metadata.

That's a good idea. This would make the `version` field redundant/not necessary right?
One thing to note is that the asset name labels allow you to also define other NFT types with different metadata structures. For instance the `222` NFT Metadata Standard is mostly about "Collectibles" or media NFTs. I think extensions here should be suitable, otherwise it may make sense to introduce a new asset name label.

Reply:
> @KtorZ, thank you for reviewing this CIP.

I am only starting now actually (well, ~20 minutes ago). This remark was more of a memo but I admittedly hadn't read the CIP in full when I wrote it 2 hours ago. 

>  This would make the version field redundant/not necessary right?

I think _so_. Provided that this proposal reaches a final state and that any modification / additions is done exclusively through extensions. 

> Separate to this minting policy script, there is a registry which can introduce new standards under the main minting policy

So, if I understand the idea correctly, the "registry" is a user-specified subset of an agreed upon set of standard referenced in [CIP-0067](https://github.com/cardano-foundation/CIPs/pull/298). This would therefore acts as this kind of extension mechanism that I am suggesting, right? Or do I get this completely wrong and simply am reading what I would want to be reading :)?

Reply:
>So, if I understand the idea correctly, the "registry" is a user-specified subset of an agreed upon set of standard referenced in https://github.com/cardano-foundation/CIPs/pull/298. This would therefore acts as this kind of extension mechanism that I am suggesting, right? Or do I get this completely wrong and simply am reading what I would want to be reading :)?

Yes that is correct. The question is just do we need for every little change/addition in the metadata an entire new asset name label standard. I think a combination of your idea with extensions within the same standard and having asset name labels to define new structures and token types is the best.

Reply:
@KtorZ how would you continue with this extension idea? Do you think it's overkill when we already have CIP-0067 to register token types with different metadata formats? Wouldn't the extensions need their own standard/registry?